test/CMakeLists.txt

```diff
@@ -30,12 +30,16 @@ endif()
 
 function(add_spvtools_unittest)
   if (NOT "${SPIRV_SKIP_TESTS}" AND TARGET gmock_main)
-    set(one_value_args TARGET)
+    set(one_value_args TARGET PCH_FILE)
     set(multi_value_args SRCS LIBS ENVIRONMENT)
     cmake_parse_arguments(
       ARG "" "${one_value_args}" "${multi_value_args}" ${ARGN})
     set(target test_${ARG_TARGET})
-    add_executable(${target} ${ARG_SRCS})
+    set(SRC_COPY ${ARG_SRCS})
+    if (DEFINED ARG_PCH_FILE)
+      PCH(SRC_COPY ${ARG_PCH_FILE})
+    endif()
+    add_executable(${target} ${SRC_COPY})
```

Comment:
I couldn't figure out a way to do this without making a copy (SRC_COPY = ${ARG_SRCS}). But I'm still a novice at cmake, and there's probably a way to avoid this. Any suggestions?

Reply:
I'm a novice with cmake as well, but why can't you just use `${ARG_SRCS}` in both places you use `${SRC_COPY}`?

Reply:
I think it is because ARG_SRCS is a "multi value" thing (a collection of source file names), and I need to actually turn it into a single variable that is a list. I tried a few other things and none worked.

Reply:
I looked into this some more. I think the issue is that the PCH macro appends to the list that is its first argument, and you can't do that using a reference to the function parameter. So this makes a bit more sense now.